### ISSUE TITLE
Inconsistent introspection results [Warning: Large GIF]

### ISSUE THREAD

**Issue by jibiabraham** [hasura/graphql-engine#2607]
When I run a graphql introspection query with variables in it, the results are inconsistent.

GIF: Inconsistent results
![out2](https://user-images.githubusercontent.com/1381646/61960288-d145e800-afe2-11e9-9df4-fbf88e5f15d9.gif)

Same query without variables returns the right results
GIF: Right results
![out3](https://user-images.githubusercontent.com/1381646/61960598-647f1d80-afe3-11e9-9292-04384483c707.gif)

**Comment by 0x777** [comment]
@jibiabraham It is hard to follow a GIF to understand the bug. Can you post the schema, metadata and the queries for which the results are not as expected?

**Comment by jibiabraham** [comment]
@0x777 I'll add a reproducible case asap. As it stands, I'm getting different results from Hasura for the same query, randomly. That's what I was trying to show with the GIF

**Comment by marionschleifer** [comment]
@jibiabraham any follow here? 🙂

**Comment by marionschleifer** [comment]
@jibiabraham I'm closing this issue. Feel free to re-open if you would like to add more information 🙂
